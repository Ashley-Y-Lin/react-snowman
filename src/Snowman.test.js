import { render, fireEvent } from "@testing-library/react";
import Snowman from "./Snowman";
import "./Snowman.css";
import img0 from "./0.png";
import img1 from "./1.png";
import img2 from "./2.png";
import img3 from "./3.png";
import img4 from "./4.png";
import img5 from "./5.png";
import img6 from "./6.png";

it("renders the Snowman without crashing", function () {
  render(
    <Snowman
      photos={[img0, img1, img2, img3, img4, img5, img6]}
      words={["apple"]}
      maxWrong={6}
    />
  );
});

it("displays 'you lose', 'answer' when maxWrong is reached, and doesn't show the buttons", function () {
  const word = "apple";
  const maxWrong = 6;

  const { container } = render(
    <Snowman
      photos={[img0, img1, img2, img3, img4, img5, img6]}
      words={[word]}
      maxWrong={maxWrong}
    />
  );

    const buttons = container.querySelectorAll("button");
    let wrongGuessNum = 0;

    for (let i=0; wrongGuessNum<maxWrong; i++){
      if (word.includes(buttons[i].value ))continue;
      
      wrongGuessNum++;
      fireEvent.click(buttons[i]);
    }

  // expect the button area to have visibility hidden
  expect(
    container.querySelector('.buttonsArea')
  ).toHaveStyle('visibility: hidden')

  // expect the message "You lose" to be displayed
  expect(
    container.querySelector('.messageArea')
  ).toContainHTML("You lose");

  // expect the correct word to be displayed
  expect(
    container.querySelector('.messageArea')
  ).toContainHTML("apple");

  // expect it to match snapshot
  expect(container).toMatchSnapshot();
});