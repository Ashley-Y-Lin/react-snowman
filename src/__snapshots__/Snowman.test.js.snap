// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays 'you lose' when maxWrong is reached 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="buttonsArea"
      style="visibility: hidden;"
    >
      <button
        value="a"
      >
        a
      </button>
      <button
        disabled=""
        value="b"
      >
        b
      </button>
      <button
        disabled=""
        value="c"
      >
        c
      </button>
      <button
        disabled=""
        value="d"
      >
        d
      </button>
      <button
        value="e"
      >
        e
      </button>
      <button
        disabled=""
        value="f"
      >
        f
      </button>
      <button
        disabled=""
        value="g"
      >
        g
      </button>
      <button
        disabled=""
        value="h"
      >
        h
      </button>
      <button
        value="i"
      >
        i
      </button>
      <button
        value="j"
      >
        j
      </button>
      <button
        value="k"
      >
        k
      </button>
      <button
        value="l"
      >
        l
      </button>
      <button
        value="m"
      >
        m
      </button>
      <button
        value="n"
      >
        n
      </button>
      <button
        value="o"
      >
        o
      </button>
      <button
        value="p"
      >
        p
      </button>
      <button
        value="q"
      >
        q
      </button>
      <button
        value="r"
      >
        r
      </button>
      <button
        value="s"
      >
        s
      </button>
      <button
        value="t"
      >
        t
      </button>
      <button
        value="u"
      >
        u
      </button>
      <button
        value="v"
      >
        v
      </button>
      <button
        value="w"
      >
        w
      </button>
      <button
        value="x"
      >
        x
      </button>
      <button
        value="y"
      >
        y
      </button>
      <button
        value="z"
      >
        z
      </button>
    </p>
    <h3
      class="messageArea"
    >
      \`You lose, the correct word was 
      apple
      \`
    </h3>
  </div>
</div>
`;

exports[`displays 'you lose',  when maxWrong is reached 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="buttonsArea"
      style="visibility: hidden;"
    >
      <button
        value="a"
      >
        a
      </button>
      <button
        disabled=""
        value="b"
      >
        b
      </button>
      <button
        disabled=""
        value="c"
      >
        c
      </button>
      <button
        disabled=""
        value="d"
      >
        d
      </button>
      <button
        value="e"
      >
        e
      </button>
      <button
        disabled=""
        value="f"
      >
        f
      </button>
      <button
        disabled=""
        value="g"
      >
        g
      </button>
      <button
        disabled=""
        value="h"
      >
        h
      </button>
      <button
        value="i"
      >
        i
      </button>
      <button
        value="j"
      >
        j
      </button>
      <button
        value="k"
      >
        k
      </button>
      <button
        value="l"
      >
        l
      </button>
      <button
        value="m"
      >
        m
      </button>
      <button
        value="n"
      >
        n
      </button>
      <button
        value="o"
      >
        o
      </button>
      <button
        value="p"
      >
        p
      </button>
      <button
        value="q"
      >
        q
      </button>
      <button
        value="r"
      >
        r
      </button>
      <button
        value="s"
      >
        s
      </button>
      <button
        value="t"
      >
        t
      </button>
      <button
        value="u"
      >
        u
      </button>
      <button
        value="v"
      >
        v
      </button>
      <button
        value="w"
      >
        w
      </button>
      <button
        value="x"
      >
        x
      </button>
      <button
        value="y"
      >
        y
      </button>
      <button
        value="z"
      >
        z
      </button>
    </p>
    <h3
      class="messageArea"
    >
      \`You lose, the correct word was 
      apple
      \`
    </h3>
  </div>
</div>
`;

exports[`displays 'you lose', ' when maxWrong is reached 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="buttonsArea"
      style="visibility: hidden;"
    >
      <button
        value="a"
      >
        a
      </button>
      <button
        disabled=""
        value="b"
      >
        b
      </button>
      <button
        disabled=""
        value="c"
      >
        c
      </button>
      <button
        disabled=""
        value="d"
      >
        d
      </button>
      <button
        value="e"
      >
        e
      </button>
      <button
        disabled=""
        value="f"
      >
        f
      </button>
      <button
        disabled=""
        value="g"
      >
        g
      </button>
      <button
        disabled=""
        value="h"
      >
        h
      </button>
      <button
        value="i"
      >
        i
      </button>
      <button
        value="j"
      >
        j
      </button>
      <button
        value="k"
      >
        k
      </button>
      <button
        value="l"
      >
        l
      </button>
      <button
        value="m"
      >
        m
      </button>
      <button
        value="n"
      >
        n
      </button>
      <button
        value="o"
      >
        o
      </button>
      <button
        value="p"
      >
        p
      </button>
      <button
        value="q"
      >
        q
      </button>
      <button
        value="r"
      >
        r
      </button>
      <button
        value="s"
      >
        s
      </button>
      <button
        value="t"
      >
        t
      </button>
      <button
        value="u"
      >
        u
      </button>
      <button
        value="v"
      >
        v
      </button>
      <button
        value="w"
      >
        w
      </button>
      <button
        value="x"
      >
        x
      </button>
      <button
        value="y"
      >
        y
      </button>
      <button
        value="z"
      >
        z
      </button>
    </p>
    <h3
      class="messageArea"
    >
      \`You lose, the correct word was 
      apple
      \`
    </h3>
  </div>
</div>
`;

exports[`displays 'you lose', 'answer' when maxWrong is reached 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="buttonsArea"
      style="visibility: hidden;"
    >
      <button
        value="a"
      >
        a
      </button>
      <button
        disabled=""
        value="b"
      >
        b
      </button>
      <button
        disabled=""
        value="c"
      >
        c
      </button>
      <button
        disabled=""
        value="d"
      >
        d
      </button>
      <button
        value="e"
      >
        e
      </button>
      <button
        disabled=""
        value="f"
      >
        f
      </button>
      <button
        disabled=""
        value="g"
      >
        g
      </button>
      <button
        disabled=""
        value="h"
      >
        h
      </button>
      <button
        value="i"
      >
        i
      </button>
      <button
        value="j"
      >
        j
      </button>
      <button
        value="k"
      >
        k
      </button>
      <button
        value="l"
      >
        l
      </button>
      <button
        value="m"
      >
        m
      </button>
      <button
        value="n"
      >
        n
      </button>
      <button
        value="o"
      >
        o
      </button>
      <button
        value="p"
      >
        p
      </button>
      <button
        value="q"
      >
        q
      </button>
      <button
        value="r"
      >
        r
      </button>
      <button
        value="s"
      >
        s
      </button>
      <button
        value="t"
      >
        t
      </button>
      <button
        value="u"
      >
        u
      </button>
      <button
        value="v"
      >
        v
      </button>
      <button
        value="w"
      >
        w
      </button>
      <button
        value="x"
      >
        x
      </button>
      <button
        value="y"
      >
        y
      </button>
      <button
        value="z"
      >
        z
      </button>
    </p>
    <h3
      class="messageArea"
    >
      \`You lose, the correct word was 
      apple
      \`
    </h3>
  </div>
</div>
`;

exports[`displays 'you lose', 'answer' when maxWrong is reached,  1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="buttonsArea"
      style="visibility: hidden;"
    >
      <button
        value="a"
      >
        a
      </button>
      <button
        disabled=""
        value="b"
      >
        b
      </button>
      <button
        disabled=""
        value="c"
      >
        c
      </button>
      <button
        disabled=""
        value="d"
      >
        d
      </button>
      <button
        value="e"
      >
        e
      </button>
      <button
        disabled=""
        value="f"
      >
        f
      </button>
      <button
        disabled=""
        value="g"
      >
        g
      </button>
      <button
        disabled=""
        value="h"
      >
        h
      </button>
      <button
        value="i"
      >
        i
      </button>
      <button
        value="j"
      >
        j
      </button>
      <button
        value="k"
      >
        k
      </button>
      <button
        value="l"
      >
        l
      </button>
      <button
        value="m"
      >
        m
      </button>
      <button
        value="n"
      >
        n
      </button>
      <button
        value="o"
      >
        o
      </button>
      <button
        value="p"
      >
        p
      </button>
      <button
        value="q"
      >
        q
      </button>
      <button
        value="r"
      >
        r
      </button>
      <button
        value="s"
      >
        s
      </button>
      <button
        value="t"
      >
        t
      </button>
      <button
        value="u"
      >
        u
      </button>
      <button
        value="v"
      >
        v
      </button>
      <button
        value="w"
      >
        w
      </button>
      <button
        value="x"
      >
        x
      </button>
      <button
        value="y"
      >
        y
      </button>
      <button
        value="z"
      >
        z
      </button>
    </p>
    <h3
      class="messageArea"
    >
      \`You lose, the correct word was 
      apple
      \`
    </h3>
  </div>
</div>
`;

exports[`displays 'you lose', 'answer' when maxWrong is reached, and  1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="buttonsArea"
      style="visibility: hidden;"
    >
      <button
        value="a"
      >
        a
      </button>
      <button
        disabled=""
        value="b"
      >
        b
      </button>
      <button
        disabled=""
        value="c"
      >
        c
      </button>
      <button
        disabled=""
        value="d"
      >
        d
      </button>
      <button
        value="e"
      >
        e
      </button>
      <button
        disabled=""
        value="f"
      >
        f
      </button>
      <button
        disabled=""
        value="g"
      >
        g
      </button>
      <button
        disabled=""
        value="h"
      >
        h
      </button>
      <button
        value="i"
      >
        i
      </button>
      <button
        value="j"
      >
        j
      </button>
      <button
        value="k"
      >
        k
      </button>
      <button
        value="l"
      >
        l
      </button>
      <button
        value="m"
      >
        m
      </button>
      <button
        value="n"
      >
        n
      </button>
      <button
        value="o"
      >
        o
      </button>
      <button
        value="p"
      >
        p
      </button>
      <button
        value="q"
      >
        q
      </button>
      <button
        value="r"
      >
        r
      </button>
      <button
        value="s"
      >
        s
      </button>
      <button
        value="t"
      >
        t
      </button>
      <button
        value="u"
      >
        u
      </button>
      <button
        value="v"
      >
        v
      </button>
      <button
        value="w"
      >
        w
      </button>
      <button
        value="x"
      >
        x
      </button>
      <button
        value="y"
      >
        y
      </button>
      <button
        value="z"
      >
        z
      </button>
    </p>
    <h3
      class="messageArea"
    >
      \`You lose, the correct word was 
      apple
      \`
    </h3>
  </div>
</div>
`;

exports[`displays 'you lose', 'answer' when maxWrong is reached, and doesn't show the buttons 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="buttonsArea"
      style="visibility: hidden;"
    >
      <button
        value="a"
      >
        a
      </button>
      <button
        disabled=""
        value="b"
      >
        b
      </button>
      <button
        disabled=""
        value="c"
      >
        c
      </button>
      <button
        disabled=""
        value="d"
      >
        d
      </button>
      <button
        value="e"
      >
        e
      </button>
      <button
        disabled=""
        value="f"
      >
        f
      </button>
      <button
        disabled=""
        value="g"
      >
        g
      </button>
      <button
        disabled=""
        value="h"
      >
        h
      </button>
      <button
        value="i"
      >
        i
      </button>
      <button
        value="j"
      >
        j
      </button>
      <button
        value="k"
      >
        k
      </button>
      <button
        value="l"
      >
        l
      </button>
      <button
        value="m"
      >
        m
      </button>
      <button
        value="n"
      >
        n
      </button>
      <button
        value="o"
      >
        o
      </button>
      <button
        value="p"
      >
        p
      </button>
      <button
        value="q"
      >
        q
      </button>
      <button
        value="r"
      >
        r
      </button>
      <button
        value="s"
      >
        s
      </button>
      <button
        value="t"
      >
        t
      </button>
      <button
        value="u"
      >
        u
      </button>
      <button
        value="v"
      >
        v
      </button>
      <button
        value="w"
      >
        w
      </button>
      <button
        value="x"
      >
        x
      </button>
      <button
        value="y"
      >
        y
      </button>
      <button
        value="z"
      >
        z
      </button>
    </p>
    <h3
      class="messageArea"
    >
      \`You lose, the correct word was 
      apple
      \`
    </h3>
  </div>
</div>
`;

exports[`displays 'you lose', 'answer' when maxWrong is reached, and doesnt show the buttons 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="buttonsArea"
      style="visibility: hidden;"
    >
      <button
        value="a"
      >
        a
      </button>
      <button
        disabled=""
        value="b"
      >
        b
      </button>
      <button
        disabled=""
        value="c"
      >
        c
      </button>
      <button
        disabled=""
        value="d"
      >
        d
      </button>
      <button
        value="e"
      >
        e
      </button>
      <button
        disabled=""
        value="f"
      >
        f
      </button>
      <button
        disabled=""
        value="g"
      >
        g
      </button>
      <button
        disabled=""
        value="h"
      >
        h
      </button>
      <button
        value="i"
      >
        i
      </button>
      <button
        value="j"
      >
        j
      </button>
      <button
        value="k"
      >
        k
      </button>
      <button
        value="l"
      >
        l
      </button>
      <button
        value="m"
      >
        m
      </button>
      <button
        value="n"
      >
        n
      </button>
      <button
        value="o"
      >
        o
      </button>
      <button
        value="p"
      >
        p
      </button>
      <button
        value="q"
      >
        q
      </button>
      <button
        value="r"
      >
        r
      </button>
      <button
        value="s"
      >
        s
      </button>
      <button
        value="t"
      >
        t
      </button>
      <button
        value="u"
      >
        u
      </button>
      <button
        value="v"
      >
        v
      </button>
      <button
        value="w"
      >
        w
      </button>
      <button
        value="x"
      >
        x
      </button>
      <button
        value="y"
      >
        y
      </button>
      <button
        value="z"
      >
        z
      </button>
    </p>
    <h3
      class="messageArea"
    >
      \`You lose, the correct word was 
      apple
      \`
    </h3>
  </div>
</div>
`;

exports[`displays 'you lose', answer when maxWrong is reached 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="buttonsArea"
      style="visibility: hidden;"
    >
      <button
        value="a"
      >
        a
      </button>
      <button
        disabled=""
        value="b"
      >
        b
      </button>
      <button
        disabled=""
        value="c"
      >
        c
      </button>
      <button
        disabled=""
        value="d"
      >
        d
      </button>
      <button
        value="e"
      >
        e
      </button>
      <button
        disabled=""
        value="f"
      >
        f
      </button>
      <button
        disabled=""
        value="g"
      >
        g
      </button>
      <button
        disabled=""
        value="h"
      >
        h
      </button>
      <button
        value="i"
      >
        i
      </button>
      <button
        value="j"
      >
        j
      </button>
      <button
        value="k"
      >
        k
      </button>
      <button
        value="l"
      >
        l
      </button>
      <button
        value="m"
      >
        m
      </button>
      <button
        value="n"
      >
        n
      </button>
      <button
        value="o"
      >
        o
      </button>
      <button
        value="p"
      >
        p
      </button>
      <button
        value="q"
      >
        q
      </button>
      <button
        value="r"
      >
        r
      </button>
      <button
        value="s"
      >
        s
      </button>
      <button
        value="t"
      >
        t
      </button>
      <button
        value="u"
      >
        u
      </button>
      <button
        value="v"
      >
        v
      </button>
      <button
        value="w"
      >
        w
      </button>
      <button
        value="x"
      >
        x
      </button>
      <button
        value="y"
      >
        y
      </button>
      <button
        value="z"
      >
        z
      </button>
    </p>
    <h3
      class="messageArea"
    >
      \`You lose, the correct word was 
      apple
      \`
    </h3>
  </div>
</div>
`;

exports[`displays you lose when maxWrong is reached 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="buttonsArea"
      style="visibility: hidden;"
    >
      <button
        value="a"
      >
        a
      </button>
      <button
        disabled=""
        value="b"
      >
        b
      </button>
      <button
        disabled=""
        value="c"
      >
        c
      </button>
      <button
        disabled=""
        value="d"
      >
        d
      </button>
      <button
        value="e"
      >
        e
      </button>
      <button
        disabled=""
        value="f"
      >
        f
      </button>
      <button
        disabled=""
        value="g"
      >
        g
      </button>
      <button
        disabled=""
        value="h"
      >
        h
      </button>
      <button
        value="i"
      >
        i
      </button>
      <button
        value="j"
      >
        j
      </button>
      <button
        value="k"
      >
        k
      </button>
      <button
        value="l"
      >
        l
      </button>
      <button
        value="m"
      >
        m
      </button>
      <button
        value="n"
      >
        n
      </button>
      <button
        value="o"
      >
        o
      </button>
      <button
        value="p"
      >
        p
      </button>
      <button
        value="q"
      >
        q
      </button>
      <button
        value="r"
      >
        r
      </button>
      <button
        value="s"
      >
        s
      </button>
      <button
        value="t"
      >
        t
      </button>
      <button
        value="u"
      >
        u
      </button>
      <button
        value="v"
      >
        v
      </button>
      <button
        value="w"
      >
        w
      </button>
      <button
        value="x"
      >
        x
      </button>
      <button
        value="y"
      >
        y
      </button>
      <button
        value="z"
      >
        z
      </button>
    </p>
    <h3
      class="messageArea"
    >
      \`You lose, the correct word was 
      apple
      \`
    </h3>
  </div>
</div>
`;
